packages/Monorepo/src/RepositoryToPackageMerger.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types=1);
+
+namespace Symplify\Monorepo;
+
+use GitWrapper\GitWorkingCopy;
+use GitWrapper\GitWrapper;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symplify\Monorepo\Filesystem\Filesystem;
+use Symplify\Monorepo\Worker\MoveHistoryWorker;
+use Symplify\Monorepo\Worker\RepositoryWorker;
+
+final class RepositoryToPackageMerger
+{
+    /**
+     * @var RepositoryWorker
+     */
+    private $repositoryWorker;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var SymfonyStyle
+     */
+    private $symfonyStyle;
+
+    /**
+     * @var GitWrapper
+     */
+    private $gitWrapper;
+
+    /**
+     * @var MoveHistoryWorker
+     */
+    private $moveHistoryWorker;
+
+    public function __construct(
+        GitWrapper $gitWrapper,
+        RepositoryWorker $repositoryWorker,
+        Filesystem $filesystem,
+        SymfonyStyle $symfonyStyle,
+        MoveHistoryWorker $moveHistoryWorker
+    ) {
+        $this->gitWrapper = $gitWrapper;
+        $this->repositoryWorker = $repositoryWorker;
+        $this->filesystem = $filesystem;
+        $this->symfonyStyle = $symfonyStyle;
+        $this->moveHistoryWorker = $moveHistoryWorker;
+    }
+
+    public function mergeRepositoryToPackage(
+        string $repositoryUrl,
+        string $monorepoDirectory,
+        string $packageSubdirectory
+    ): void {
+        $gitWorkingCopy = $this->getGitWorkingCopyForDirectory($monorepoDirectory);
+
+        // add repository as remote and merge
+        $this->repositoryWorker->mergeRepositoryToMonorepoDirectory($repositoryUrl, $gitWorkingCopy);
+        $this->symfonyStyle->success(sprintf(
```

Comment:
It would be better in command, but I try to decouple this whole services from there to lower coupling to Symfony Console.

On the other hand, here is coupling to Symfony Console via SymfonyStyle, but I don't know how to decouple it better. Any ideas? I'd like to have this as clean as possible :+1: